File: 977. Squares of a Sorted Array/977. Squares of a Sorted Array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortedSquares = function (nums) {
  const squaredSorted = [];
  const n = nums.length;
  let left = 0;
  let right = n - 1;

  while (left <= right) {
    const leftSquared = nums[left] ** 2;
    const rightSquared = nums[right] ** 2;

    if (leftSquared > rightSquared) {
      squaredSorted.push(leftSquared);
      left++;
    } else {
      squaredSorted.push(rightSquared);
      right--;
    }
  }

  squaredSorted.reverse();

  return squaredSorted;
};

File: 977. Squares of a Sorted Array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        n = len(nums)
        left = 0
        right = n - 1

        ans = []

        while left <= right:
            if abs(nums[left]) > abs(nums[right]):
                ans.append(nums[left] ** 2)
                left += 1
            else:
                ans.append(nums[right] ** 2)
                right -= 1

        return ans[::-1]
